Can't compile on 32-bit ARM

Seems like there's a bug that causes compilation to fail on 32-bit systems:

```
$ cargo build --all
    Updating crates.io index
   Compiling libc v0.2.137
   Compiling byteorder v1.4.3
   Compiling positioned-io v0.3.0 (/home/user/temp/positioned-io)
error[E0308]: mismatched types
    --> src/raf.rs:68:54
     |
68   |             libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as i64, libc::POSIX_FADV_RANDOM);
     |             -------------------                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `i64`
     |             |
     |             arguments to this function are incorrect
     |
note: function defined here
    --> /home/user/.cargo/registry/src/github.com-1285ae84e5963aae/libc-0.2.137/src/unix/linux_like/mod.rs:1655:12
     |
1655 |     pub fn posix_fadvise(fd: ::c_int, offset: ::off_t, len: ::off_t, advise: ::c_int) -> ::c_int;
     |            ^^^^^^^^^^^^^
help: you can convert an `i64` to an `i32` and panic if the converted value doesn't fit
     |
68   |             libc::posix_fadvise(file.as_raw_fd(), 0, (file.metadata()?.len() as i64).try_into().unwrap(), libc::POSIX_FADV_RANDOM);
     |                                                      +                             +++++++++++++++++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `positioned-io` due to previous error
```

--- a/src/raf.rs
+++ b/src/raf.rs
@@ -65,6 +65,9 @@ impl RandomAccessFile {
     fn try_new_impl(file: File) -> io::Result<RandomAccessFile> {
         unsafe {
             use std::os::unix::io::AsRawFd;
+            #[cfg(target_pointer_width = "32")]
+            libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as i32, libc::POSIX_FADV_RANDOM);
+            #[cfg(target_pointer_width = "64")]
             libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as i64, libc::POSIX_FADV_RANDOM);
         }
 

--- a/src/raf.rs
+++ b/src/raf.rs
@@ -65,10 +65,7 @@ impl RandomAccessFile {
     fn try_new_impl(file: File) -> io::Result<RandomAccessFile> {
         unsafe {
             use std::os::unix::io::AsRawFd;
-            #[cfg(target_pointer_width = "32")]
-            libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as i32, libc::POSIX_FADV_RANDOM);
-            #[cfg(target_pointer_width = "64")]
-            libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as i64, libc::POSIX_FADV_RANDOM);
+            libc::posix_fadvise(file.as_raw_fd(), 0, file.metadata()?.len() as libc::off_t, libc::POSIX_FADV_RANDOM);
         }
 
         Ok(RandomAccessFile { file })

